src/python/pants/option/compiler_option_sets_mixin.py

```diff
@@ -0,0 +1,99 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from builtins import object
+
+
+class CompilerOptionSetsMixin(object):
+  """A mixin for language-scoped that support compiler option sets."""
+
+  @classmethod
+  def register_options(cls, register):
+    super(CompilerOptionSetsMixin, cls).register_options(register)
+
+    register('--fatal-warnings-enabled-args', advanced=True, type=list, fingerprint=True,
```

Comment:
The `--fatal-warnings-*` options should probably be deprecated (we can calculate everything from `compiler_option_sets`)? I'm not sure if there's a reason to leave them in -- @stuhood is there something I'm missing?

Reply:
I left that in for deprecation purposes. Will add a deprecation target version.

Reply:
Cool, that's all, thanks!